### PR TITLE
Go Parser Integration Tests

### LINKED ISSUE #5741: Go Parser: Integration test parity with Python parser tests
## Background ##
The Python parser has a comprehensive integration test suite under tdrs-backend/tdpservice/parsers/test/ that validates end-to-end parsing behavior across all program types, sections, and edge cases. The Go parser needs equivalent integration test coverage to ensure it produces identical results before migration can proceed. The goal is to keep all the test logic inside of the Python test environment, validate the results of the Go parser, and avoid setting up a complicated integration test suite inside of the Go parser.

### Acceptance Criteria ###
- [ ] Inventory of all Python parser integration tests in tdrs-backend/tdpservice/parsers/test/
- [ ] Equivalent Go integration tests covering the same scenarios
- [ ] Tests validate parsed record output matches expected values
- [ ] Tests validate parser error output matches expected errors (count, category, message)
- [ ] Tests cover all program types: TANF, SSP, Tribal TANF, FRA, TANF Program Audit
- [ ] Tests cover all sections for each program type
- [ ] Tests cover edge cases: empty files, malformed headers, encrypted fields, duplicate records
- [ ] Tests use the same test fixture files as Python tests where possible
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

### Notes ###
- Python test directory: tdrs-backend/tdpservice/parsers/test/
- This is a critical gate for migration confidence

### Open Questions ###
- None

### PR DESCRIPTION
**Pull request by elipe17**
## Summary of Changes
This PR implements a large subset of the python test_parse.py integration tests from the perspective of the Go parser. I left TODOs for remaining tests and justifications inline with the tests if the number of errors changed between the go parser and python parser. Follow on issues need to be generated to finish the remaining tests. This PR is too large already to do any more.

- Added new integration test suite for pytest
- Added custom settings and logic to run new go integration tests
- Adding missing validators for validators.yaml and to appropriate schema defs throughout to satisfy currently enabled tests
- Added new go parser compliant fixtures for certain python parse tests
- Register go parser task in python so python celery side knows about it
- Update go parser to consume from separate queue name than all other python celery tasks
- Update Factories to use UUID based emails since the go parser uses a shared state postgres DB
- Update README to indicate how to run go parser integration tests
- Generalized go parser cat4 validators to work with any record type
- Removed duplicate field on SSP M1 record
- Add header ctx for validators that need to know about the header when doing any kind of validation
- Add sentinel errors for unknown record type and others
- Created custom Redis backend based on gocelery interface to indicate to celery when a go parser task has actually completed

Pull request closes #5741 

## How to Test

```
cd tdrs-frontend && docker-compose up --build
cd tdrs-backend && docker-compose up --build
```

1. Run `task backend-pytest-go-integration` and the standard pytest unit tests in parallel. Assert that they dont conflict and that they all pass
2. Run them multiple times and ensure the still pass
3. Review the code changes and identify any issues :)

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [ ] Inventory of all Python parser integration tests in tdrs-backend/tdpservice/parsers/test/
- [ ] Equivalent Go integration tests covering the same scenarios
- [ ] Tests validate parsed record output matches expected values
- [ ] Tests validate parser error output matches expected errors (count, category, message)
- [ ] Tests cover all program types: TANF, SSP, Tribal TANF, FRA, TANF Program Audit
- [ ] Tests cover all sections for each program type
- [ ] Tests cover edge cases: empty files, malformed headers, encrypted fields, duplicate records
- [ ] Tests use the same test fixture files as Python tests where possible
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: tdrs-backend/tdpservice/parsers/test/integration/test_go_parse.py [hunk 1]
```diff
@@ -0,0 +1,1871 @@
...
+    dfs.refresh_from_db()
+    return dfs
+
+
+@pytest.mark.go_parser_integration
+@pytest.mark.usefixtures("go_parser_datafile_cleanup")
+class TestGoParse:
+    """Tests for parse and validation flows."""
+
+    # @pytest.fixture
```
Comment (jtimpe) [inline tdrs-backend/tdpservice/parsers/test/integration/test_go_parse.py:103]: are we intending on using this first block of commented-out tests?

File: .circleci/build-and-test/jobs.yml [hunk 2]
```diff
@@ -15,6 +15,12 @@
           command: |
             cd tdrs-backend;
             docker-compose run --rm web bash -c "./wait_for_services.sh && pytest --cov-report=xml"
+      - run:
+          name: Run Go parser integration tests
+          command: |
+            cd tdrs-backend
+            docker compose up --build -d go-parser
+            docker compose -f docker-compose.yml exec web sh -c "DJANGO_SETTINGS_MODULE=tdpservice.settings.go_parser_integration DJANGO_CONFIGURATION=GoParserIntegration pytest -c pytest-go-parser.ini --basetemp=/home/tdpapp/ tdpservice/parsers/test/integration"
```
Comment (jtimpe) [inline .circleci/build-and-test/jobs.yml:23]: can we use the `backend-pytest-go-integration` task here so we don't have to update the command in multiple cases (if it ever changes)?

**Review by jtimpe — APPROVED** [review]
Looking good!

File: tdrs-backend/tdpservice/parsers/test/integration/test_go_parse.py (hunk 1 above)
Comment (elipe17) [reply tdrs-backend/tdpservice/parsers/test/integration/test_go_parse.py:103]: Most likely. I left them commented out because I felt there were more important tests lower down that gave us a better bang for the buck.

File: .circleci/build-and-test/jobs.yml (hunk 2 above)
Comment (elipe17) [reply .circleci/build-and-test/jobs.yml:23]: Is task installed for these processes? The only reason I didnt use task is because I was lazy didn't want to worry about installing it haha.

**Comment by raftmsohani** [comment]
Looks good!

**Review by raftmsohani — APPROVED** [review]

File: .circleci/build-and-test/jobs.yml (hunk 2 above)
Comment (jtimpe) [reply .circleci/build-and-test/jobs.yml:23]: it is! i used `task` for the commands in #5803. we don't use it anywhere else currently, though
